### PR TITLE
[IMP] l10n_sg: adds new tax rates for 2024

### PR DESCRIPTION
**Pull request by tong-odoo**
SG Government has announced to change the GST-rate on 01 Jan 2024 MN from 8% to 9%

Changes:
- Added new taxes (8% -> 9%)
- Added tax groups
- Added migration script to l10n_sg to apply those changes

task-3454556

### CODE REVIEW

File: addons/l10n_sg/data/l10n_sg_chart_data.xml [hunk 1]
```diff
@@ -316,47 +316,47 @@
         </record>
 
         <record model="account.account.template" id="account_account_738">
-            <field name="name">Purchase Tax Account 8% TX-N33</field>
+            <field name="name">Purchase Tax Account 9% TX-N33</field>
```
Comment (vin-odoo) [inline addons/l10n_sg/data/l10n_sg_chart_data.xml:319]: The only uncertainty I have with this pr is that renaming.
In 16.2+, I think the update script will also update the account name (not 100% sure) but I think it should be checked too.
Because in this case, it would not be very consistent until 2024 right.

File: addons/l10n_sg/data/account_tax_data_2024.xml
Comment (vin-odoo) [inline addons/l10n_sg/data/account_tax_data_2024.xml:1]: account_tax_**template**_data_2024

File: addons/l10n_sg/data/account_data.xml [hunk 2]
```diff
@@ -13,5 +13,11 @@
             <field name="name">TAX 8%</field>
         </record>
 
+        <!-- Account Tax Group (post-2024 rates change) -->
+        <record id="tax_group_9" model="account.tax.group">
+            <field name="name">TAX 9%</field>
+        </record>
+
```
Comment (vin-odoo) [inline addons/l10n_sg/data/account_data.xml:20]: May as well put this in the new file no?

**Review by vin-odoo — COMMENTED** [review]
Mostly good to go :+1:

**Comment by vin-odoo** [comment]
@clbr-odoo Hi Claire! I know it's a pretty simple PR but as the expert in tax update it would be great if you could do a quick check and let me know if everything looks fine to you :smile: 

tldr:
- Taxes need to be added for 2024, we need to keep the old 8% ones for a bit as they could still be used for ongoing deals.
- We also renamed the accounts to be less linked to the tax rates, we know that this won't affect the existing database until 16.2

File: addons/l10n_sg/data/account_data.xml [hunk 3]
```diff
@@ -14,4 +14,5 @@
         </record>
 
     </data>
+
```
Comment (vin-odoo) [inline addons/l10n_sg/data/account_data.xml:17]:
```suggestion

```
Please remove this line to avoid unnecessary diff

**Review by vin-odoo — CHANGES_REQUESTED** [review]
Small oopsie, after that we're good to go once Claire gives her OK

**Comment by clbr-odoo** [comment]
Hello @vin-odoo  !
Seems all good, you got it right for the tax name change that won't be effective before saas-16.2 for existing databases :+1:

**Review by vin-odoo — APPROVED** [review]
@robodoo r+

Thanks for the great job Tommy!
